src/com/carolinarollergirls/scoreboard/core/impl/TeamImpl.java

```diff
@@ -85,6 +86,14 @@ protected Object computeValue(PermanentProperty prop, Object value, Object last,
             return t.isRunning() && this == t.getOwner() && t.isReview();
         }
         if (prop == Value.TRIP_SCORE && flag != Flag.COPY && scoreBoard.isInJam() && (Integer)value > 0) {
+            
+            // If points arrive during an initial trip and we are not in overtime, assign the points to the first scoring trip instead.
+            if (getCurrentTrip().getNumber() == 1 && !getScoreBoard().isInOvertime()) {
+                set(Value.TRIP_SCORE,0);
+                getRunningOrEndedTeamJam().addScoringTrip();
```

Comment:
This should be `execute(ADD_TRIP)` in order to take care of setting lost if appropriate.

And I think the two calls to set are noops.

Reply:
The two calls to set are not noops - when a score arrives, the first call resets the score on the first score trip back to zero.

A new trip is added (which I've updated to use ADD_TRIP, thanks for the heads up.)

Then, the second call to set sets the value of THAT trip to the incoming score.  I've tested this, and it works with those two commands and fails without them.

However - this version DOES fail the test suite, and the reason for that is because the test suite assigns points WITHOUT assigning a trip.  Our test case, in other words, is the exact situation this fix is designed to prevent, so no wonder it breaks.  Desired behavior would be:

points entered -> scoreboard immediately assigns points to second trip, marks lead as LOST.
lead subsequently pressed -> stats change to show jammer lost the ability to become lead, but lead is not displayed.

The test suite should be updated so that it does NOT expect lead to be displayed in this case, but it could also be modified in other ways, so I won't immediately change it, but ask what you think the best modification should be to match the code change.

Reply:
As far as the internal data goes - I disagree that having the timestamp wrong on the end of the initial trip is sufficient reason to flag the user, and nothing else in THIS case is incorrect.  The user may be handling the cases wrong where the not-lead jammer fails to score points, but that is not relevant to this code change, which ONLY trips when they do score points.

Reply:
> I've tested this, and it works with those two commands and fails without them.

I've tried to reproduce this test as this is counter to my understanding of how the code should behave. (The function here is called before any value is changed. So updating current trip in it should cause the points to be written to the new trip right away.)

Result after a few iterations of initially confusing results: If the points are entered during a Jam they go to the scoring trip even without the `set` calls. If they are added after the Jam they go to the initial trip even with those calls and no scoring trip is added at all. Once I found the pattern, I also found the reason: You placed this inside an `if` that has `isInJam` as a condition. Should be a straightforward fix.

Reply:
Fixed in subsequent commits.